File: app/src/main/java/lt/vilnius/tvarkau/entity/ReportType.java
package lt.vilnius.tvarkau.entity;

import android.os.Parcel;
import android.os.Parcelable;

public class ReportType implements Parcelable {

    private String name;

    public ReportType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
    }


    protected ReportType(Parcel in) {
        this.name = in.readString();
    }

    public static final Parcelable.Creator<ReportType> CREATOR = new Parcelable.Creator<ReportType>() {
        @Override
        public ReportType createFromParcel(Parcel source) {
            return new ReportType(source);
        }

        @Override
        public ReportType[] newArray(int size) {
            return new ReportType[size];
        }
    };
}

File: app/src/main/java/lt/vilnius/tvarkau/entity/ReportType.kt
package lt.vilnius.tvarkau.entity

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Entity(
    tableName = "report_types"
)
@Parcelize
data class ReportType(
    @PrimaryKey
    @ColumnInfo(name = "id")
    val id: Int,
    @ColumnInfo(name = "title")
    val title: String
) : Parcelable
